infra/stream/InputStream.hpp

```diff
@@ -63,13 +63,53 @@ namespace infra
         StreamErrorPolicy& errorPolicy;
     };
 
+    template<class Parent, class TheReader>
```

Comment:
Can we find a better name for TheReader? E.g. ReaderType?

Reply:
Sounds good, I'll apply it